### PR TITLE
docs: add the release process

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@riseup.net>

### CODE REVIEW

File: docs/release.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+***************
+Release Process
+***************
+
+As this project is critical to the Python ecosystem's supply chain security, all
+releases are PGP signed with one of the keys listed in the `installation page`_.
+Before releasing please make sure you PGP key is listed there, and preferably
```
Comment (henryiii) [inline docs/release.rst:7]:
```suggestion
Before releasing please make sure your PGP key is listed there, and preferably
```

File: docs/release.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+signed by one of the other key holders. If your key is not signed my one of the
```
Comment (henryiii) [inline docs/release.rst:8]:
```suggestion
signed by one of the other key holders. If your key is not signed by one of the
```

File: docs/release.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+other key holders, please make sure the PR that added your key to the
+`installation page`_ was approved by at least one other maintainer.
+
+After that is done, you may release the project by following these steps:
+
+#. Bump the versions in ``setup.cfg`` and ``src/build/__init__.py``
```
Comment (henryiii) [inline docs/release.rst:14]: Reminder, needs to be updated after #471.

File: docs/release.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+#. Update ``CHANGELOG.rst`` with the new version and current date
+#. Make a release commit with the changes made above
+  - The commit message should follow the ``release X.Y.Z`` format
+#. Make a signed tag (``git tag -s X.Y.Z``)
+  - The tag title should follow the ``build X.Y.Z`` format
+  - The tag body should be a plaintext version of the changelog for the current
+    release
+#. Push the commit and tag to the repository (``git push`` and ``git push --tags``)
+#. Build the Python artifacts (``python -m build``)
```
Comment (henryiii) [inline docs/release.rst:23]: Very meta. :D

**Review by henryiii — APPROVED** [review]

**Review by layday — APPROVED** [review]
Thanks!

File: docs/release.rst (hunk 3 above)
Comment (FFY00) [reply docs/release.rst:14]: I merged it now, I am gonna update this PR.

**Pull request merged**